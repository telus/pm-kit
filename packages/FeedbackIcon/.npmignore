*.stories.{js,mdx}
rollup.config.js
FeedbackIcon.jsx
__tests__/