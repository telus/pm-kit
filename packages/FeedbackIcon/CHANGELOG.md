# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [2.0.3](https://github.com/telus/pm-kit/compare/@pm-kit/feedback-icon@2.0.2...@pm-kit/feedback-icon@2.0.3) (2020-11-13)

**Note:** Version bump only for package @pm-kit/feedback-icon





## [2.0.2](https://github.com/telus/pm-kit/compare/@pm-kit/feedback-icon@2.0.1...@pm-kit/feedback-icon@2.0.2) (2020-09-29)


### Bug Fixes

* **checkbox:** Update error message font styles ([#73](https://github.com/telus/pm-kit/issues/73)) ([37cec91](https://github.com/telus/pm-kit/commit/37cec91c91010a8a3e32ddd67ef5e449e17c9ae4))





## [2.0.1](https://github.com/telus/pm-kit/compare/@pm-kit/feedback-icon@2.0.0...@pm-kit/feedback-icon@2.0.1) (2020-03-17)


### Bug Fixes

* **feedback-icon:** updated styling  ([#62](https://github.com/telus/pm-kit/issues/62)) ([9800415](https://github.com/telus/pm-kit/commit/980041551fedc99598ba8b1aad4f5045fec4b50e))





# [2.0.0](https://github.com/telus/pm-kit/compare/@pm-kit/feedback-icon@1.0.0...@pm-kit/feedback-icon@2.0.0) (2020-01-29)


### Features

* **feedback-icon:** renamed state options ([#45](https://github.com/telus/pm-kit/issues/45)) ([197bf12](https://github.com/telus/pm-kit/commit/197bf12da7aab2f2c3424e2ec982229678677c7c))


### BREAKING CHANGES

* **feedback-icon:** options for `state` prop are now "success", "error", and "waiting", from previously "pass", "failed", "waiting"





# 1.0.0 (2020-01-22)

### Features

- **feedback-icon:** added FeedbackIcon package ([#26](https://github.com/telus/pm-kit/issues/26)) ([bd79b5e](https://github.com/telus/pm-kit/commit/bd79b5e6c372680c08000cf83242ce1b9bdc3158))
