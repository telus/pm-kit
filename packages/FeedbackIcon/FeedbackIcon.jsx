import React from 'react'
import { css } from '@emotion/core'
import { parkGreen } from '@pm-kit/colours'
import PropTypes from 'prop-types'
// import { successPath, failPath } from '../../shared/svg/index'

const FeedbackIcon = ({ state, size }) => {
  const loader = css`
    @keyframes spinner {
      to {
        transform: rotate(360deg);
      }
    }
    border: 3px solid ${parkGreen};
    border-radius: 50%;
    border-right-color: transparent;
    display: inline-block;
    width: ${size};
    height: ${size};
    animation: 1s linear infinite spinner;
    animation-delay: var(--spinner-delay);
  `

  const imageStyles = css`
    width: ${size};
    height: ${size};
  `

  return (
    <>
      {state !== 'disabled' && (
        <>
          {state === 'waiting' && <div css={loader}></div>}
          {/* {state === 'passed' && <img css={imageStyles} src={successPath} alt="success" />}
          {state === 'failed' && <img css={imageStyles} src={failPath} alt="fail" />} */}
        </>
      )}
    </>
  )
}

FeedbackIcon.propTypes = {
  state: PropTypes.oneOf(['waiting', 'passed', 'failed', 'disabled']),
  /**
   * The height and width of feedback-icon.
   */
  size: PropTypes.string,
}

FeedbackIcon.defaultProps = {
  state: 'disabled',
  size: '1rem',
}

export default FeedbackIcon
