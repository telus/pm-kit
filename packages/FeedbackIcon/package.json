{
  "name": "@pm-kit/feedback-icon",
  "version": "2.0.0",
  "main": "dist/index.cjs.js",
  "module": "dist/index.es.js",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "author": "Public Mobile",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telus/public-mobile-design-system/issues"
  },
  "peerDependencies": {
    "@emotion/core": "^10.0.22",
    "react": "^16.12.0",
    "react-dom": "^16.12.0"
  },
  "dependencies": {
    "@pm-kit/colours": "^1.1.0",
    "prop-types": "^15.7.2"
  }
}
