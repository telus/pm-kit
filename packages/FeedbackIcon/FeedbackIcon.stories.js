import React from 'react'
import FeedbackIcon from './FeedbackIcon'
import { withKnobs, select, text } from '@storybook/addon-knobs'

export default {
  title: 'Design System|FeedbackIcon',
  component: FeedbackIcon,
  decorators: [withKnobs],
}

export const passed = () => <FeedbackIcon state="passed" />

export const failed = () => <FeedbackIcon state="failed" />

export const waiting = () => <FeedbackIcon state="waiting" />

export const disabled = () => <FeedbackIcon state="disabled" />

export const Playground = () => {
  return (
    <FeedbackIcon
      state={select('state', ['passed', 'failed', 'waiting', 'disabled'], 'disabled')}
      size={text('size', '1.125rem')}
    />
  )
}

Playground.story = {
  name: 'playground',
  parameters: { docs: { page: null, disable: true } },
}
