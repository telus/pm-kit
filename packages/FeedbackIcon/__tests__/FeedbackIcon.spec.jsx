import React from 'react'
import { render } from 'enzyme'

import FeedbackIcon from '../FeedbackIcon'

describe('FeedbackIcon', () => {
  it('renders nothing when disabled', () => {
    const feedbackIcon = render(<FeedbackIcon state="disabled" />)
    expect(feedbackIcon).toMatchSnapshot()
  })

  it('renders checkmark when passed', () => {
    const feedbackIcon = render(<FeedbackIcon state="passed" />)
    expect(feedbackIcon).toMatchSnapshot()
  })

  it('renders error icon when failed', () => {
    const feedbackIcon = render(<FeedbackIcon state="failed" />)
    expect(feedbackIcon).toMatchSnapshot()
  })

  it('renders spinner when waiting', () => {
    const feedbackIcon = render(<FeedbackIcon state="waiting" />)
    expect(feedbackIcon).toMatchSnapshot()
  })
})
