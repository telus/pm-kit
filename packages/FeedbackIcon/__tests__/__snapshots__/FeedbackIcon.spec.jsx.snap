// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackIcon renders checkmark when passed 1`] = `
.emotion-0 {
  width: 1rem;
  height: 1rem;
}

<img
  alt="passed"
  class="emotion-0"
  src="test-file-stub"
/>
`;

exports[`FeedbackIcon renders error icon when failed 1`] = `
.emotion-0 {
  width: 1rem;
  height: 1rem;
}

<img
  alt="failed"
  class="emotion-0"
  src="test-file-stub"
/>
`;

exports[`FeedbackIcon renders nothing when disabled 1`] = `null`;

exports[`FeedbackIcon renders spinner when waiting 1`] = `
.emotion-0 {
  border: 3px solid #034045;
  border-radius: 50%;
  border-right-color: transparent;
  display: inline-block;
  width: 1rem;
  height: 1rem;
  -webkit-animation: 1s linear infinite spinner;
  animation: 1s linear infinite spinner;
  -webkit-animation-delay: var(--spinner-delay);
  animation-delay: var(--spinner-delay);
}

<div
  class="emotion-0"
/>
`;
